vm/state_migration/src/type_migrations/init/state_v9_to_v10.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use fil_actor_init_v10::State as InitStateV10;
+use fil_actor_init_v9::State as InitStateV9;
+use fil_actors_runtime_v10::{make_map_with_root, Map};
+use forest_shim::{
+    address::{Address, PAYLOAD_HASH_LEN},
+    state_tree::ActorID,
+};
+use fvm_ipld_blockstore::Blockstore;
+
+use crate::common::{TypeMigration, TypeMigrator};
+
+impl<BS> TypeMigration<BS, InitStateV9, InitStateV10> for TypeMigrator
```

Comment:
Or maybe this could go to `fvm_shim` or `fil-actor-states`? @lemmih @LesnyRumcajs

Reply:
I think it would go well in `fil-actor-states`. Might be better to keep it in Forest until we've finalized the design, though.

Reply:
Personally, I do believe it belongs to the `state_migration` crate and be kept private. Unless you think something outside state migrations would find it useful?

Reply:
I'm not sure about this, but the non-blockstore parts of the migration may be helpful for other projects. For example, there could be a `From` (or `TryFrom`) implementation for `MinerInfoV10` -> `MinerInfoV11`.

Reply:
No strong opinion here. I'm in favour of not exposing things because once it's part of the public interface, we need to maintain it and can't modify it without major version bumps which are seldom appreciated by the community.